cmd/ak/cmd/triggers/create.go

```diff
@@ -52,41 +53,33 @@ var createCmd = common.StandardCommand(&cobra.Command{
 			eid = e.ID()
 		}
 
-		// Mode 1: connection is required, event and/or filter
-		// and/or data are required, and schedule is not allowed.
-		c, cid, err := r.ConnectionNameOrID(ctx, connection, project)
-		if connection != "" {
-			if err = common.AddNotFoundErrIfCond(err, c.IsValid()); err != nil {
-				return common.ToExitCodeErrorNotNilErr(err, "connection")
-			}
-		}
-
-		// Mode 2: schedule is required, connection/event/filter are not allowed.
-		// TODO(ENG-1004): Verify validity of schedule expression.
-		if schedule != "" {
-			cid = sdktypes.BuiltinSchedulerConnectionID
-			event = fixtures.SchedulerEventTriggerType
-			data[fixtures.ScheduleExpression] = schedule
-		}
-
-		// Finally, create and print the trigger.
-		m := kittehs.TransformMapValues(data, func(v string) *sdktypes.ValuePB {
-			return sdktypes.ToProto(sdktypes.NewStringValue(v))
-		})
-
 		t, err := sdktypes.StrictTriggerFromProto(&sdktypes.TriggerPB{
 			Name:         name,
 			EnvId:        eid.String(),
-			ConnectionId: cid.String(),
 			EventType:    event,
 			Filter:       filter,
-			Data:         m,
 			CodeLocation: cl.ToProto(),
+			Schedule:     schedule,
 		})
 		if err != nil {
 			return fmt.Errorf("invalid trigger: %w", err)
 		}
 
+		if connection != "" {
+			_, cid, err := r.ConnectionNameOrID(ctx, connection, project)
+			if err != nil {
+				return common.ToExitCodeErrorNotNilErr(err, "connection")
+			}
+
+			t = t.WithConnectionID(cid)
+		} else if schedule != "" {
+			t = t.WithSchedule(schedule)
+		} else if webhook {
+			t = t.WithWebhook()
+		} else {
+			return fmt.Errorf("missing connection, schedule or webhook")
+		}
```

Comment:
These flags are mutually exclusive. Use separate `if` blocks instead of a single block with `else` - the code will look cleaner.

Reply:
i think i disagree with you here. i think this is more self documenting.

Reply:
i changed my mind. fixed.